projects/plugins/social/src/class-social-shares.php

```diff
@@ -0,0 +1,88 @@
+<?php
+/**
+ * Register the Social note custom post type.
+ *
+ * @package automattic/jetpack-social-plugin
+ */
+
+namespace Automattic\Jetpack\Social;
+
+/**
+ * Register the Jetpack Social Shares Class.
+ */
+class Social_Shares {
+	const SOCIAL_SHARES_POST_META_KEY = '_publicize_shares';
+
+	/**
+	 * Return a list of of social shares.
+	 *
+	 * @param int $post_id The Post ID.
+	 *
+	 * @return array
+	 */
+	public static function get_social_shares( $post_id ) {
+		if ( empty( $post_id ) ) {
+			return array();
+		}
+
+		$shares = get_post_meta( $post_id, self::SOCIAL_SHARES_POST_META_KEY );
+
+		if ( empty( $shares ) ) {
+			return array();
+		}
+
+		return array_filter(
+			$shares[0],
+			function ( $share ) {
+				return isset( $share['status'] ) && $share['status'] === 'success';
+			}
+		);
+	}
+
+	/**
+	 * Return a html to display the social shares.
+	 *
+	 * @param int $post_id The Post ID.
+	 *
+	 * @return array
+	 */
+	public static function get_the_social_shares( $post_id ) {
+		$shares = self::get_social_shares( $post_id );
+
+		if ( empty( $shares ) ) {
+			return '<span></span>';
+		}
+
+		$shares_by_service = array();
+
+		foreach ( $shares as $share ) {
+			$service   = $share['service'];
+			$timestamp = $share['timestamp'];
+
+			if ( ! isset( $shares_by_service[ $service ] ) || $timestamp > $shares_by_service[ $service ]['timestamp'] ) {
+				$shares_by_service[ $service ] = $share;
+			}
+		}
+
+		$links = array();
+		foreach ( $shares_by_service as $service => $item ) {
+			$links[] = '<a href="' . $item['message'] . '">' . $service . '</a>';
+		}
+
+		$text = implode( ', ', $links );
+		if ( count( $links ) > 1 ) {
+			$last_link = array_pop( $links );
+			return 'Also on ' . implode( ', ', $links ) . " and $last_link";
```

Comment:
I wonder if the 'Also on' should be the default, but we could make it configurable in some way. Do you think it would also be worth adding some markup? Maybe make it a header, or wrap it in a `span`?

The links are also essentially a list, so maybe we should mark it up as one.

Reply:
I have added a custom filter hook to do any manipulations the user may want. Do you think that solves this? 

I did think of a list, but I thought they were all in the same line and thought to have wrapped it in a span element.